CRM/Core/Payment.php

```diff
@@ -261,7 +261,14 @@ public static function logPaymentNotification($params) {
     }
 
     $log = new CRM_Utils_SystemLogger();
-    $log->alert($message, $_REQUEST);
+    // $_REQUEST doesn't handle JSON, to support providers that POST JSON we need the raw POST data.
```

Comment:
What about checking the in put type first - per https://www.stevenrombauts.be/2018/06/read-json-request-data-with-php/